src/pymortests/reductors/loewner.py

```diff
@@ -0,0 +1,69 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+import numpy as np
+import pytest
+import scipy.sparse as sps
+
+from pymor.models.iosys import LTIModel
+from pymor.reductors.loewner import LoewnerReductor
+
+pytestmark = pytest.mark.builtin
+
+np.random.seed(0)
+custom_partitioning = np.random.permutation(10)
+
+test_data = [
+    ({'r': 3}, {}),
+    ({'tol': 1e-3}, {}),
+    ({'tol': 1e-3}, {'partitioning': 'even-odd'}),
+    ({'tol': 1e-3}, {'partitioning': 'half-half'}),
+    ({'tol': 1e-3}, {'partitioning': (custom_partitioning[:5], custom_partitioning[5:])}),
+    ({'tol': 1e-3}, {'ordering': 'magnitude'}),
+    ({'tol': 1e-3}, {'ordering': 'random'}),
+    ({'tol': 1e-3}, {'ordering': 'regular'}),
+    ({'tol': 1e-3}, {'conjugate': False}),
+    ({'tol': 1e-3}, {'mimo_handling': 'full'}),
+    ({'tol': 1e-3}, {'mimo_handling': 'random'}),
+    ({'tol': 1e-3}, {'mimo_handling': (np.random.rand(10, 3), np.random.rand(2, 10))})
+]
+
+
+@pytest.mark.parametrize('reduce_kwargs,loewner_kwargs', test_data)
+def test_loewner_lti(reduce_kwargs, loewner_kwargs):
+    fom = make_fom(10)
+    s = np.logspace(-2, 2, 10)*1j
+    loewner = LoewnerReductor(s, fom, **loewner_kwargs)
+    rom = loewner.reduce(**reduce_kwargs)
+    assert isinstance(rom, LTIModel)
+
+
+@pytest.mark.parametrize('reduce_kwargs,loewner_kwargs', test_data)
+def test_loewner_tf(reduce_kwargs, loewner_kwargs):
+    fom = make_fom(10)
+    s = np.logspace(-2, 2, 10)*1j
+    loewner = LoewnerReductor(s, fom.transfer_function, **loewner_kwargs)
+    rom = loewner.reduce(**reduce_kwargs)
+    assert isinstance(rom, LTIModel)
+
+
+@pytest.mark.parametrize('reduce_kwargs,loewner_kwargs', test_data)
+def test_loewner_data(reduce_kwargs, loewner_kwargs):
+    fom = make_fom(10)
+    s = np.logspace(-2, 2, 10)
+    Hs = fom.transfer_function.freq_resp(s)
+    loewner = LoewnerReductor(s*1j, Hs, **loewner_kwargs)
+    rom = loewner.reduce(**reduce_kwargs)
+    assert isinstance(rom, LTIModel)
```

Comment:
I think it would be good to add tests for checking ROM order and interpolation of the FOM when `tol=0`.

Reply:
I made some adjustments to the tests, partially based on the comment above. Let me know if this is what you had in mind.